archiver_test.go

```diff
@@ -8,23 +8,33 @@ import (
 	"testing"
 )
 
-func TestZipAndUnzip(t *testing.T) {
-	symmetricTest(t, ".zip", Zip, Unzip)
+func TestArchiver(t *testing.T) {
+	for name, ar := range SupportedFormats {
+		name, ar := name, ar
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			// skip RAR for now
+			if _, ok := ar.(r); ok {
+				t.Skip("not supported")
+			}
+			symmetricTest(t, name, ar)
+		})
+	}
 }
 
 // symmetricTest performs a symmetric test by using mf to make an archive
 // from the test corpus, then using of to open the archive and comparing
 // the contents to ensure they are equal.
```

Comment:
The godoc here needs slight updating so it refers to `ar` instead of `mf` and `of`.